app/models/concerns/project/state_validator.rb

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# This module handles with default state validation
+module Project::StateValidator
+  extend ActiveSupport::Concern
+
+  included do
+
+    # All valid states for projects in_analysis to end of publication
+    ON_ANALYSIS_TO_END_STATES = %w(in_analysis approved online successful waiting_funds failed).freeze
```

Comment:
Yeah, I just kept this code here, because I think it's easier to first merge and then we refactor